### PR TITLE
Introduce dynamic handler

### PR DESCRIPTION
**Pull request by mandarjog**
1. Add `dynamic.Handler` to the dynamic package
2. Add `adapter.RemoteXXX` interfaces that dynamic handler implements
3. Update spybackend to be usable from outside.

### CODE REVIEW

File: mixer/pkg/protobuf/yaml/dynamic/handler.go [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+	// Svc encapsulates abstract service
+	Svc struct {
+		Name       string
+		Pkg        string
+		MethodName string
+		InputType  string
+		OutputType string
+
+		encoder *messageEncoder
```
Comment (guptasu) [inline mixer/pkg/protobuf/yaml/dynamic/handler.go]: consider saving everything here that we need for session based plan too. WDYT ?

File: mixer/pkg/protobuf/yaml/dynamic/handler.go [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+	}
+
+	// Instance name
+	Instance struct {
```
Comment (guptasu) [inline mixer/pkg/protobuf/yaml/dynamic/handler.go]: SvcInfo ?

File: mixer/pkg/protobuf/yaml/dynamic/handler.go [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+		n:          &atomic.Uint64{},
+	}
+
+	var svc *Svc
+	for _, inst := range instances {
+		if svc, err = RemoteAdapterSvc("Handle",
+			protoyaml.NewResolver(inst.FileDescSet), sessionBased, adapterConfig); err != nil {
+			return nil, err
+		}
+		hh.svcMap[inst.Name] = svc
```
Comment (guptasu) [inline mixer/pkg/protobuf/yaml/dynamic/handler.go]: inst.template.name

Comment (guptasu) [reply mixer/pkg/protobuf/yaml/dynamic/handler.go]: infact there will be multiple instances of the same template.. so we need to group and dispatch after trouping.

**Comment by mandarjog** [comment]
@guptasu everything except saving things for session based. That will be done in a separate PR.

**Pull request merged**